app/controllers/devise/registrations_controller.rb

```diff
@@ -63,11 +63,11 @@ def update
 
   # DELETE /resource
   def destroy
-    resource.destroy
-    Devise.sign_out_all_scopes ? sign_out : sign_out(resource_name)
-    set_flash_message! :notice, :destroyed
-    yield resource if block_given?
-    respond_with_navigational(resource){ redirect_to after_sign_out_path_for(resource_name) }
+    if destroy_resource(resource, account_destroy_params)
+      sign_out_resource(resource_name, resource)
```

Comment:
The spec is failing because you aren't passing the block argument down, probably doing this will fix the spec:

```ruby
def destroy(&block)
  ...
  sign_out_resource(resource_name, resource, &block) # << here you pass the block down
  ...
end
```